egg-herbie/main.rkt

```diff
@@ -215,6 +217,38 @@
 ;; egraph -> string
 (define-eggmath egraph_serialize (_fun _egraph-pointer -> _rust/datum))
 
+;; egraph -> uint
+(define-eggmath egraph_size (_fun _egraph-pointer -> _uint))
+
+;; egraph -> id -> uint
+(define-eggmath egraph_eclass_size (_fun _egraph-pointer _uint -> _uint))
+
+;; egraph -> id -> idx -> uint
+(define-eggmath egraph_enode_size (_fun _egraph-pointer _uint _uint -> _uint))
+
+;; egraph -> u32vector
+(define-eggmath
+ egraph_get_eclasses
+ (_fun [e : _egraph-pointer] [v : _u32vector = (make-u32vector (egraph_size e))] -> _void -> v))
+
+;; egraph -> id -> u32 -> (values (or symbol number) u32vector)
+(define-eggmath egraph_get_node
+                (_fun [e : _egraph-pointer]
+                      [id : _uint32]
+                      [idx : _uint32]
+                      [v : _u32vector = (make-u32vector (egraph_enode_size e id idx))]
```

Comment:
So this will allocate for each node. I think that's fine but you could imagine actually streaming the enodes over during analysis so that we only allocate one u32 for each size of enode.

Reply:
Ah right. I think I do a similar trick elsewhere.

Reply:
Yes, that's what I thought about. I think, separate PR, but an idea.